src/ngraph/op/fused/normalize_l2.cpp

```diff
@@ -0,0 +1,114 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <iterator>
+
+#include "ngraph/builder/norm.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/divide.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/util/broadcasting.hpp"
+#include "normalize_l2.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+const string op::NormalizeL2::type_name{"NormalizeL2"};
+
+op::NormalizeL2::NormalizeL2(const Output<Node>& data,
+                             const Output<Node>& axes,
+                             float eps,
+                             EpsMode eps_mode)
+    : FusedOp({data, axes})
+    , m_eps{eps}
+    , m_eps_mode{eps_mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::NormalizeL2::pre_validate_and_infer_types()
+{
+    const auto& data_pshape = get_input_partial_shape(0);
+    const auto& axes_pshape = get_input_partial_shape(1);
+
+    const Shape data_shape{data_pshape.to_shape()};
```

Comment:
Does it work for cases with dynamic input shape?

Reply:
You are right - 'to_shape' throw exception when argument has dynamic shape. I added additional assertion.